### ISSUE TITLE
Socket request: Cannot read property 'headers' of undefined

### ISSUE THREAD

**Issue by Oldwo1f** [balderdashy/sails#3015]
I am trying to reach a controller method via socket but the request is interceped by something.
using http i reach this method normaly **(but i cannot do my realtime stuff)**

* Request never arrive to my method
* I checked all my policies, request never arrived here too

on client side i recieved this error, but i couldnt figured out from where 

```
Cannot read property 'headers' of undefined
```

i am using 
* "sails": "git://github.com/balderdashy/sails.git",
* "socket.io-redis": "^0.1.4",
* socket io client side : // socket.io-1.2.1

**Comment by sgress454** [comment]
@Oldwo1f It would help if you could figure out where the error is coming from.  Usually in the Javascript console a log message indicates the line that generated it.  In Chrome it actually provides a link, which when you click it brings you to the exact line of Javascript that triggered the log.

**Comment by sailsbot** [comment]
Thanks for posting, @Oldwo1f.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
